src/com/books/BookStorage.java

```diff
@@ -0,0 +1,38 @@
+package com.books;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.Set;
+
+public class BookStorage extends ArrayList<Book> {
+
+    public BookStorage filterByAuthorName(String name) {
```

Comment:
It's good idea to give filtering responsibility to another class, but it seems strange. Why does this class extend ArrayList? 
What if we will move our book storage to a relational database? And then to NoSQL DB? Or to the just plain text storage?